src/modules/uORB/uORBDeviceMaster.cpp

```diff
@@ -403,39 +379,37 @@ void uORB::DeviceMaster::showTop(char **topic_filter, int num_filters)
 uORB::DeviceNode *uORB::DeviceMaster::getDeviceNode(const char *nodepath)
 {
 	lock();
-	uORB::DeviceNode *node = getDeviceNodeLocked(nodepath);
-	unlock();
-	//We can safely return the node that can be used by any thread, because
-	//a DeviceNode never gets deleted.
-	return node;
-}
-
-
-#ifdef __PX4_NUTTX
-uORB::DeviceNode *uORB::DeviceMaster::getDeviceNodeLocked(const char *nodepath)
-{
-	uORB::DeviceNode *rc = nullptr;
 
-	if (_node_map.find(nodepath)) {
-		rc = _node_map.get(nodepath);
+	for (DeviceNode *node = _node_list.getHead(); node != nullptr; node = node->getSibling()) {
+		if (strcmp(node->get_devname(), nodepath) == 0) {
+			unlock();
+			return node;
+		}
 	}
 
-	return rc;
-}
+	unlock();
 
-#else
+	return nullptr;
+}
 
-uORB::DeviceNode *uORB::DeviceMaster::getDeviceNodeLocked(const char *nodepath)
+uORB::DeviceNode *uORB::DeviceMaster::getDeviceNode(const struct orb_metadata *meta, const uint8_t instance)
 {
-	uORB::DeviceNode *rc = nullptr;
-	std::string np(nodepath);
+	lock();
+	uORB::DeviceNode *node = getDeviceNodeLocked(meta, instance);
+	unlock();
 
-	auto iter = _node_map.find(np);
+	//We can safely return the node that can be used by any thread, because
+	//a DeviceNode never gets deleted.
+	return node;
+}
 
-	if (iter != _node_map.end()) {
-		rc = iter->second;
+uORB::DeviceNode *uORB::DeviceMaster::getDeviceNodeLocked(const struct orb_metadata *meta, const uint8_t instance)
+{
+	for (DeviceNode *node = _node_list.getHead(); node != nullptr; node = node->getSibling()) {
+		if ((node->get_name() == meta->o_name) && (node->get_instance() == instance)) {
```

Comment:
`node->get_name() == meta->o_name` is a bit dangerous. It works but I'd rather either compare the strings or compare the meta pointer.